pkg/migrations/sql/idempotent/telemetry.sql

```diff
@@ -0,0 +1,94 @@
+-- promscale_telemetry_housekeeping() does telemetry housekeeping stuff, which includes
+-- searching the table once an hour, looking for stale promscales, if found,
+-- adding their values into the counter_reset row, and then cleaning up the stale
+-- promscale instances.
+-- It is concurrency safe, since it takes lock on the promscale_instance_information table,
+-- making sure that at one time, only one housekeeping is being done.
+-- 
+-- It returns TRUE if last run was beyond an hour, otherwise FALSE.
+CREATE OR REPLACE FUNCTION SCHEMA_PS_CATALOG.promscale_telemetry_housekeeping()
+RETURNS BOOLEAN AS
+$$
+    DECLARE
+        is_last_run_beyond_hour BOOLEAN := FALSE;
+    BEGIN
+        LOCK TABLE SCHEMA_PS_CATALOG.promscale_instance_information IN ACCESS EXCLUSIVE MODE NOWAIT; -- Do not wait for the lock as some promscale is already cleaning up the stuff.
```

Comment:
Do we need to lock the whole table or just the counter reset row?

Reply:
I think we should lock the whole table, otherwise, while some promscale is doing the counter reset updation stuff, some other promscale will update its values, as updation of respective rows happen before the go function that locks is called, so it seems a race.